### PR TITLE
Rm preflow payments

### PR DESCRIPTION
**Pull request by seelabs**
This removes the old payment engine from rippled. The changes in RippleCalc should be carefully reviewed. The commits are split as they are so they can be easily reviewed. The first commit removes the engine, but does as little refactoring/reformatting as possible. The other commits clean up the code a bit.

### CODE REVIEW

File: src/ripple/app/paths/RippleCalc.cpp [hunk 1]
```diff
@@ -100,17 +114,17 @@ RippleCalc::Output RippleCalc::rippleCalculate (
...
-            ", dstAmtReq: " << saDstAmountReq <<
-            ", sendMax: " << saMaxAmountReq <<
-            ", algo: " << "V2";
+    j.debug() << "RippleCalc Result> "
+              << " actualIn: " << flowOut.actualAmountIn
+              << ", actualOut: " << flowOut.actualAmountOut
+              << ", result: " << flowOut.result()
+              << ", dstAmtReq: " << saDstAmountReq
+              << ", sendMax: " << saMaxAmountReq << ", algo: "
+              << "V2";
```
Comment (scottschurr) [inline src/ripple/app/paths/RippleCalc.cpp]: Is the "algo: V2" still relevant?

**Comment by miguelportilla** [comment]
Due to the changes, the following comments may no longer apply:

https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/ripple/ledger/View.h#L302

https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/ripple/protocol/jss.h#L180
https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/ripple/protocol/jss.h#L239
https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/ripple/protocol/jss.h#L276
https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/ripple/protocol/jss.h#L535

https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/ripple/protocol/jss.h#L265
https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/ripple/protocol/jss.h#L376

**Comment by miguelportilla** [comment]
The loop can be reduce to a single offer this part of the test:

https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/test/app/Offer_test.cpp#L273-L275

eg.

```     ...
        env (pay (gw, alice, USD (100)));
        env (pay (gw, carol, EUR (100)));
        env (offer (carol, USD (1), EUR (2)));```

**Comment by miguelportilla** [comment]
`tecExpect` may no longer be needed...

https://github.com/seelabs/rippled/blob/rm-preflow-payments/src/test/app/Offer_test.cpp#L3572-L3573

File: src/ripple/app/paths/RippleCalc.cpp (hunk 1 above)
Comment (seelabs) [reply src/ripple/app/paths/RippleCalc.cpp]: I have some old scripts that use it, but on re-consideration, those scripts are obsolete now. Removed.

**Comment by seelabs** [comment]
@miguelportilla Nice finds, all. I addressed them all, except I kept the loop in the regression test (with a comment).

**Pull request closed without merging**